client/src/components/settings/toggle-button-setting.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useRef } from 'react';
+import ToggleCheck from '../../assets/icons/toggle-check';
+import '../helpers/toggle-button.css';
+import './toggle-setting.css';
+
+type ToggleButtonSettingProps = {
+  action: string;
+  explain?: string;
+  flag: boolean;
+  toggleFlag: () => void;
+  offLabel: string;
+  onLabel: string;
+};
+
+export default function ToggleButtonSetting({
+  action,
+  explain,
+  flag,
+  toggleFlag,
+  ...restProps
+}: ToggleButtonSettingProps): JSX.Element {
+  const randomId = Math.floor(Math.random() * 1_000_000);
```

Comment:
Ahh, thanks for pointing this out. I think the reason I didn't use `flagName` in the first place is because originally they were not unique. If you look at the donation toggle you'll see that it was originally using `showPortfolio` for the `flagName` by mistake, thus there were duplicate `flagName` values. Even though I fixed that and they are all unique now, I must have forgot about it and decided to create a random ID instead to guarantee uniqueness.

Reply:
Didn't notice the comment until ahmad pushed it to the notification, sorry for late reply.

I think this is the wrong mindset to have, there is 1/100000 chance that the random function id will provide the same id, so instead of relaying on chance to provide unique description, we can make it more obvious that `flagName` should be unique:

- adding comments.
- changing the variable to `screenReaderContext`.
- providing `@example` and `@params` in TypeScript, to show as hints in the editor.

it isn't enforcing the unique `flagName`, but people will be going out of their way to ignore all of these hints, and if contributor do, members shouldn't.

---

I know that we are people that do make mistakes, I can say that because I am the most person to make mistakes, but the solution shouldn't be making an abstract solution that can be forgotten, then removed because someone didn't remember the importance of the uniqueness.

but we have to get better and do better, and provide better info through the code that this is something that should be respected for these reasons.

Reply:
@Sboonny 
> there is 1/100000 chance that the random function id will provide the same id

I don't think this is an issue. React 18 now provides a `useId` hook which guarantees a unique id, so we can ultimately use that. In the meantime, there are modules we could use instead to get the same thing, such as react-id-generator. Or we could do our own checks to make sure the id generated by `random()` doesn't exist and just try again if it does. We could also use `Cryto.getRandomValues()` instead to get even better random numbers to decrease the chances of duplicates.

My point is that these ids are being used behind the scenes merely for accessibility reasons and thus they do not need to be known by the programmer, and should not depend on the programmer to give them a value since there is a chance of programmer error. I don't think this is an abstract solution. This is a perfectly reasonable solution to ensure that accessibility is met, and in my opinion, a much cleaner solution than all of the bullets points we'd have to implement above. And the reason for generating the id would not be forgotten as we can add a comment at the source explaining why we are generating a unique id.

But in reality, the chances of a developer using the same flagName is probably close to nil and so this is not a battle worth continuing :slightly_smiling_face: Because the accessibility of these controls needs urgent fixing and I would like to see it happen ASAP, I will acquiesce and make the suggested changes so we can get this merged.

Reply:
No need, if there's a better solution that provide a granted unique I'd, we can move to them later.

I was not aware of these other options q👍